src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java

```diff
@@ -3,35 +3,45 @@
 import lombok.RequiredArgsConstructor;
 import lombok.extern.log4j.Log4j2;
 import org.folio.consortia.client.ConsortiaConfigurationClient;
+import org.folio.consortia.client.UsersClient;
 import org.folio.consortia.domain.dto.ConsortiaConfiguration;
+import org.folio.consortia.domain.dto.PermissionUser;
 import org.folio.consortia.domain.dto.Tenant;
 import org.folio.consortia.domain.dto.TenantCollection;
+import org.folio.consortia.domain.dto.User;
 import org.folio.consortia.domain.entity.TenantEntity;
+import org.folio.consortia.domain.entity.UserTenantEntity;
 import org.folio.consortia.exception.ResourceAlreadyExistException;
 import org.folio.consortia.exception.ResourceNotFoundException;
 import org.folio.consortia.repository.TenantRepository;
 import org.folio.consortia.repository.UserTenantRepository;
 import org.folio.consortia.service.ConsortiumService;
+import org.folio.consortia.service.PermissionService;
+import org.folio.consortia.service.PermissionUserService;
 import org.folio.consortia.service.TenantService;
+import org.folio.consortia.service.UserTenantService;
 import org.folio.spring.FolioExecutionContext;
 import org.folio.spring.FolioModuleMetadata;
+import org.folio.spring.scope.FolioExecutionContextSetter;
 import org.springframework.core.convert.ConversionService;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.PageRequest;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.Objects;
+import java.util.Optional;
 import java.util.UUID;
 
 import static org.folio.consortia.utils.HelperUtils.checkIdenticalOrThrow;
-import static org.folio.consortia.utils.TenantContextUtils.createFolioExecutionContextForTenant;
-import static org.folio.consortia.utils.TenantContextUtils.runInFolioContext;
+import static org.folio.consortia.utils.TenantContextUtils.prepareContextForTenant;
 
 @Service
 @Log4j2
 @RequiredArgsConstructor
 public class TenantServiceImpl implements TenantService {
 
+  private static final String PERMISSIONS_FILE_PATH = "permissions/admin-user-permissions.csv";
```

Comment:
since we will have system user permission path soon, so lets rename to SHADOW_ADMIN_PERMISSSION_FILE_PATH